src/Integration/LocalServices/TestProjectIndicators/ProjectKindTestProjectIndicator.cs

```diff
@@ -0,0 +1,51 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Linq;
+using EnvDTE;
+
+namespace SonarLint.VisualStudio.Integration
+{
+    internal class ProjectKindTestProjectIndicator : ITestProjectIndicator
+    {
+        private readonly IProjectSystemHelper projectSystem;
+
+        public ProjectKindTestProjectIndicator(IServiceProvider serviceProvider)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            projectSystem = serviceProvider.GetService<IProjectSystemHelper>();
+            projectSystem.AssertLocalServiceIsNotNull();
+        }
+
+        public bool? IsTestProject(Project project)
+        {
+            var hierarchy = projectSystem.GetIVsHierarchy(project);
+            var aggregateProjectKinds = projectSystem.GetAggregateProjectKinds(hierarchy).ToList();
+
+            return aggregateProjectKinds.Contains(ProjectSystemHelper.TestProjectKindGuid) ||
+                   aggregateProjectKinds.Contains(ProjectSystemHelper.ExternalTestProjectKindGuid);
```

Comment:
_ExternalTestProjectKindGuid_ is the Service guid, not a project type guid.  At the MSBuild level they are represented differently:

```
<ItemGroup>
    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
</ItemGroup>
<PropertyGroup>
    <ProjectTypeGuids>{60dc8134-eba5-43b8-bcc9-bb4bc16c2548};{F184B08F-C81C-45F6-A57F-5ABD9991F28F
</ProjectTypeGuids>
 </PropertyGroup>
```

Have you manually tested that at runtime VS treats Service guids in the same way as project type guids?